File: ObjectSample/ObjectSample.scala
import scala.math._ 

trait Shape {
    def area :Double
}

object Shape {
    private class Circle(radius: Double) extends Shape{
        override val area = 3.14*radius*radius
    }

    private class Rectangle(height: Double, length: Double) extends Shape{
        override val area = height * length
    }

    private class Triangle(a: Double, b: Double, c: Double) extends Shape{
        override val area = 0.25 * math.sqrt((a+b+c)*(a+b-c)*(a+c-b)*(b+c-a))
    }
    def apply(height :Double , length :Double ) : Shape = new Rectangle(height,length)
    def apply(radius :Double) : Shape = new Circle(radius)
    def apply(a: Double, b: Double, c: Double) : Shape = new Triangle(a,b,c)
}

object Main{
    def main(args: Array[String]){
        val circle = Shape(2)
        println("Circle Area:" + circle.area)
        val rectangle = Shape(2,3)
        println("Rectange Area:" + rectangle.area)
        val triangle = Shape(4,5,3)
        println("Triagnle Area:" + triangle.area)
    }
}

File: ObjectSample/ObjectSample.java


public class ObjectSample{

    private static class Shape{
        protected double area;
        public double getArea(){
            return area;
        }
    }

    private static class Circle extends Shape{
        public Circle(double radius){
            area = 3.14*radius*radius; 
        }
    }

    private static class Rectangle extends Shape{
        public Rectangle(double width,double height){
            area = width * height;
        }
    }

    private static class Triangle extends Shape{
        public Triangle(double a,double b,double c){
            area =  0.25 * Math.sqrt((a+b+c)*(a+b-c)*(a+c-b)*(b+c-a));
        }
    }

    public static void main(String[] args){
        Shape circle = new Circle(2);
        Shape rectangle = new Rectangle(3,4);
        Shape triangle = new Triangle(3,4,5);
        System.out.println("Circle area:" + circle.getArea());
        System.out.println("Rectangle area:" + rectangle.getArea());
        System.out.println("Triangle area:" + triangle.getArea());
    }

}
